x/market/keeper/keeper.go

```diff
@@ -528,6 +528,50 @@ func (k Keeper) GetInstruments(ctx sdk.Context) (instrs []types.MarketData) {
 	return
 }
 
+// GetAllInstruments gets all instruments based on current order book. It
+// includes synthetic pairs, last order timestamp and calculates the best price
+// for the pair.
+func (k Keeper) GetAllInstruments(ctx sdk.Context) []*types.MarketData {
+	coins := k.sk.GetSupply(ctx).GetTotal().Sort()
+	n := len(coins)
+	// n instruments producing n*(n-1) pairs below
+	instrLst := make([]*types.MarketData, n*(n-1))
+	idx := 0
+	// produce cartesian products of denominations resulting in all
+	// denominations paired with each other except for themselves
+	for _, srcCoin := range coins {
+		source := srcCoin.Denom
+		for _, dstCoin := range coins {
+			destination := dstCoin.Denom
+			if source == destination {
+				continue
+			}
+
+			instrLst[idx] = &types.MarketData{
+				Source:      source,
+				Destination: destination,
+			}
+
+			// fill in best price
+			bestPlan := k.createExecutionPlan(ctx, source, destination)
```

Comment:
Looks good, but I believe we need swap the order of source, destination parameters here as createExecutionPlan creates a plan based on orders.

Otherwise I guess BestPrice will look inverted compared the LastPrice(?)

Reply:
Thank you. Resolved. But now there's a slight deviation.
```
- source: ungm
  destination: eeur
  best_price: "1.250000411854150914"
  last_price: "1.250000000000000000"
  last_traded: 2020-12-09T00:02:46.534254609Z
```
Should we truncate or round to 2..x decimals?